app/controllers/Clas.scala

```diff
@@ -206,6 +207,54 @@ final class Clas(env: Env, authC: Auth) extends LilaController(env):
       )
   }
 
+  def bulkActions(id: ClasId) = Secure(_.Teacher) { ctx ?=> me ?=>
+    WithClass(id): clas =>
+      for
+        data <- env.clas.bulk.load(clas)
+        page <- renderPage(views.clas.teacherDashboard.bulkActions(data))
+      yield Ok(page)
+  }
+
+  def bulkActionsPost(id: ClasId) = SecureBody(_.Teacher) { ctx ?=> me ?=>
+    val moveTo = """move-to-(.+)""".r
+    bindForm(lila.clas.ClasBulkForm.form)(
+      _ => Redirect(routes.Clas.bulkActions(id)).flashFailure,
+      data =>
+        data.action match
+          case "archive" =>
+            env.clas.api.student.archiveMany(data.activeUserIds.map { u => Student.makeId(u, id) }, true)
+            Redirect(routes.Clas.bulkActions(id)).flashSuccess
+          case "restore" =>
+            env.clas.api.student.archiveMany(data.archivedUserIds.map { u => Student.makeId(u, id) }, false)
```

Comment:
these 2 actions `archive` and `restore` never check that the teacher actually is in the class of the students